### ISSUE TITLE
[MLIR] couldn't convert MLIR to GraphDef via tf-mlir-translate

### ISSUE THREAD

**Issue by rusyaev-roman** [tensorflow/tensorflow#31163]
`tf-mlir-translate` fails while converting MLIR to GraphDef.

`$ uname -a`
```
Linux pc-ubuntu 4.15.0-54-generic #58-Ubuntu SMP Mon Jun 24 10:55:24 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
```

`$ cd tensorflow/compiler/mlir`
`$ cat add.pbtxt `
```
node {
  name: "Add"
  op: "Add"
  input: "input0"
  input: "input1"
  attr {
    key: "T"
    value {
      type: DT_INT32
    }
  }
}
node {
  name: "input0"
  op: "Placeholder"
  attr {
    key: "dtype"
    value {
      type: DT_INT32
    }
  }
}
node {
  name: "input1"
  op: "Placeholder"
  attr {
    key: "dtype"
    value {
      type: DT_INT32
    }
  }
}
versions {
  producer: 27
}
```

`$ bazel run tf-mlir-translate -- --graphdef-to-mlir --tf-input-arrays=input0,input1 --tf-input-data-types=DT_INT32,DT_INT32 --tf-input-shapes=10:10 --tf-output-arrays=Add ${PWD}/add.pbtxt -o ${PWD}/add.mlir`

`$ cat add.mlir`
```


module attributes {tf.versions = {bad_consumers = [], min_consumer = 0 : i32, producer = 27 : i32}} {
  func @main(%arg0: tensor<10xi32>, %arg1: tensor<10xi32>) -> tensor<10xi32>
  attributes  {tf.entry_function = {inputs = "input0, input1", outputs = "Add"}} {
    %0 = tf_executor.graph {
      %1:2 = tf_executor.island {
        %4 = "tf.Placeholder.input"(%arg0) {device = "", dtype = "tfdtype$DT_INT32", name = "input0", shape = "tfshape$dim { size: 10 }"} : (tensor<10xi32>) -> tensor<10xi32>
        tf_executor.yield %4 : tensor<10xi32>
      } {device = "", dtype = "tfdtype$DT_INT32", name = "input0", shape = "tfshape$dim { size: 10 }"}
      %2:2 = tf_executor.island {
        %4 = "tf.Placeholder.input"(%arg1) {device = "", dtype = "tfdtype$DT_INT32", name = "input1", shape = "tfshape$dim { size: 10 }"} : (tensor<10xi32>) -> tensor<10xi32>
        tf_executor.yield %4 : tensor<10xi32>
      } {device = "", dtype = "tfdtype$DT_INT32", name = "input1", shape = "tfshape$dim { size: 10 }"}
      %3:2 = tf_executor.island {
        %4 = "tf.Add"(%1#0, %2#0) {T = "tfdtype$DT_INT32", device = "", name = "Add"} : (tensor<10xi32>, tensor<10xi32>) -> tensor<10xi32>
        tf_executor.yield %4 : tensor<10xi32>
      } {T = "tfdtype$DT_INT32", device = "", name = "Add"}
      tf_executor.fetch %3#0 : tensor<10xi32>
    }
    return %0 : tensor<10xi32>
  }
}
```
`$ bazel run tf-mlir-translate -- --mlir-to-graphdef ${PWD}/add.mlir -o ${PWD}/converted_add.pbtxt`

**output**
```
...
INFO: Build completed successfully, 1 total action
INFO: Build completed successfully, 1 total action
2019-07-30 15:38:00.617166: E tensorflow/compiler/mlir/tensorflow/translate/tf_mlir_translate_registration.cc:82] Graph export failed: Failed precondition: op node 'tf_executor.graph' was not a TF op!
```

**Comment by rusyaev-roman** [comment]
CC +=  @joker-eph

**Comment by joker-eph** [comment]
Sorry about this, it was fixed in 459f8cf615402f022de9ea1f9c1f033509db0076 but reverted in 459f8cf615402f022de9ea1f9c1f033509db0076 because it broke an ASAN test. The ASAN fix is under review and we should land everything today (hopefully).

**Comment by rusyaev-roman** [comment]
Thanks!

**Comment by antiagainst** [comment]
The ASAN fix is landed as https://github.com/tensorflow/tensorflow/commit/417972d14641a9a4a3ac512a078465d6cfeaff30 now.

**Comment by joker-eph** [comment]
Fixed in ff79f54b9c89789f03b09d5a9a5afd481adca6b0
